packages/react-components/components/ValidatedTextInput.tsx

```diff
@@ -0,0 +1,82 @@
+import TextInput from '@celo/react-components/components/TextInput'
+import { validateInput } from '@celo/utils/src/inputValidation'
+import * as React from 'react'
+import { TextInputProps } from 'react-native'
+
+interface OwnProps {
+  value: string
+  numberOfDecimals?: number
+  placeholder?: string
+  lng?: string
+  onChangeText: (input: string) => void
+}
+
+// Required props when validator type is phone
+interface PhoneValidatorProps {
+  validator: 'phone'
+  countryCallingCode: string
+  // Following props unused w/ 'phone' validator but required to be defined
+  customValidator?: CustomValidatorProps['customValidator']
+}
+
+interface NumberValidatorProps {
+  validator?: 'integer' | 'decimal'
```

Comment:
A common pattern in TS is to use a `kind` property to differentiate interfaces like these:
https://github.com/celo-org/celo-monorepo/blob/master/packages/mobile/src/utils/recipient.ts#L9